src/Repository/TodoRepository.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Todo;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+
+/**
+ * @extends ServiceEntityRepository<Todo>
+ *
+ * @method Todo|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Todo|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Todo[]    findAll()
+ * @method Todo[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class TodoRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, Todo::class);
+    }
+
+    public function add(Todo $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->persist($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function remove(Todo $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->remove($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function getTodos()
+    {
+        $qb = $this->createQueryBuilder('t');
+
+        $qb->where('t.isCompleted = :isCompleted')
+            ->setParameter('isCompleted', false);
+
+        return $qb->getQuery()->getResult();
+    }
+
+    public function getTodoById($id)
+    {
+        $qb = $this->createQueryBuilder('t')
+            ->where('t.id = :id')
+            ->setParameter('id', $id);
+
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
+//    /**
+//     * @return Todo[] Returns an array of Todo objects
+//     */
+//    public function findByExampleField($value): array
+//    {
+//        return $this->createQueryBuilder('t')
+//            ->andWhere('t.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->orderBy('t.id', 'ASC')
+//            ->setMaxResults(10)
+//            ->getQuery()
+//            ->getResult()
+//        ;
+//    }
+
+//    public function findOneBySomeField($value): ?Todo
+//    {
+//        return $this->createQueryBuilder('t')
+//            ->andWhere('t.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->getQuery()
+//            ->getOneOrNullResult()
+//        ;
+//    }
+}
```

Comment:
Why do we need it?

Reply:
Pubf add and remove were created by default, pubf getTodos, getTodoById i didnt know that i need them or not so i past them for future.
I should delete them?